lib/engine_react.js

```diff
@@ -54,66 +78,38 @@ var engine_react = {
   findPartialRE: null,
 
   // render it
-  renderPattern(pattern, data, partials) {
+  renderPattern (pattern, data, partials) {
     try {
-      let runtimeCode = [];
-
-      // the all-important Babel transform, runtime version
-      runtimeCode.push(Babel.transform(moduleCodeString(pattern), {
-        presets: [ require('babel-preset-react') ],
-        plugins: [[require('babel-plugin-transform-es2015-modules-umd'), {
-          globals: {
-            "react": "React"
-          }
-        }]]
-      }).code);
+      const transpiledModule = babelTransform(pattern);
+
+      const staticMarkup = ReactDOMServer.renderToStaticMarkup(
+           React.createFactory(transpiledModule)(data)
+      );
 
       return outputTemplate.render({
-        patternPartial: pattern.patternPartial,
-        json: JSON.stringify(data),
-        htmlOutput: ReactDOMServer.renderToStaticMarkup(
-          React.createFactory(pattern.module)(data)
-        ),
-        runtimeCode: runtimeCode.join(';')
+        htmlOutput: staticMarkup
       });
     }
     catch (e) {
-	    console.log("Error rendering React pattern.", e);
-	    return "";
+      console.log('Error rendering React pattern.', e);
+      return '';
     }
   },
 
-  registerPartial(pattern) {
-    const customRequire = function (id) {
-      const registeredPattern = registeredComponents.byModuleName[id];
-      return registeredPattern ? registeredPattern.module : _require(id);
-    };
-
-    // the all-important Babel transform, server-side version
-    const compiledModule = Babel.transform(moduleCodeString(pattern), {
-      presets: [ require('babel-preset-react') ],
-      plugins: [ require('babel-plugin-transform-es2015-modules-commonjs') ]
-    });
-
+  registerPartial (pattern) {
     // add to registry
-    registeredComponents.byModuleName[pattern.patternBaseName] = pattern;
-
-    // eval() module code in this little scope that injects our
-    // custom wrap of require();
-    ((require) => {
-      /* eslint-disable no-eval */
-      pattern.module = eval(compiledModule.code);
-    })(customRequire);
+    registeredComponents.byPatternPartial[pattern.patternPartial] = pattern;
```

Comment:
This stuff that was removed also looks like it may be implicated in the client-side stuff. I'd like to be able to include modules in a very flexibly-named way. We'll have a need on my end to be able to import `atoms/buttons/Button.js` as `atoms-Button` or as `Button`. I know that's hard, but I think it's important.